### PR TITLE
Support nullable primitives in CREATE MAPPING for compact serialization [HZ-2239]

### PR DESCRIPTION
**Pull request by burakgok**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java [hunk 1]
```diff
@@ -72,4 +72,32 @@ public void smokeTest() throws Exception {
...
         instance().getSql().execute(mappingDdl2);
     }
+
+    @Test
+    public void testCompactSerialization() throws Exception {
+        Member member = instance().getCluster().getLocalMember();
+
+        instance().getMap("m1").put("foo", new CompactClass(1));
+        String mappingDdl = ((SqlClientService) client().getSql()).mappingDdl(member, "m1").get();
+        assertEquals("CREATE OR REPLACE EXTERNAL MAPPING \"hazelcast\".\"public\".\"m1\" EXTERNAL NAME \"m1\" (" + LE
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java:82]: AssertJ has a method that normalizes new lines - easier to use

**Review by k-jamroz — CHANGES_REQUESTED** [review]
Generally LGTM, but the test needs to be fixed, failed on Jenkins

File: hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java [hunk 2]
```diff
@@ -172,12 +172,15 @@ public static SortedMap<String, QueryDataType> resolveCompact(@Nonnull Schema sc
...
 
             case INT8:
+            case NULLABLE_INT8:
                 return QueryDataType.TINYINT;
 
             case INT16:
+            case NULLABLE_INT16:
                 return QueryDataType.SMALLINT;
 
             case CHAR:
```
Comment (k-jamroz) [inline hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java:186]: does compact serialization support `Character`?

Comment (k-jamroz) [reply hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java:186]: apparently not - maybe they should?

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java [hunk 3]
```diff
@@ -44,32 +43,60 @@ public static void beforeClass() {
...
+                        + "  'valueJavaClass'='com.hazelcast.jet.sql.impl.schema.model.Person'\n"
+                        + ")");
         instance().getSql().execute(mappingDdl2);
     }
+
+    @Test
+    public void testCompactSerialization() throws Exception {
+        Member member = instance().getCluster().getLocalMember();
+
+        instance().getMap("m3").put("foo", new CompactClass(1));
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java:79]: nit: usually, when we do not care, we use `randomMapName()`

**Review by k-jamroz — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java (hunk 2 above)
Comment (burakgok) [reply hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java:186]: https://hazelcast.slack.com/archives/C01JU7ZJYGP/p1681836482255999

**Pull request merged**